### ISSUE TITLE
[Bug]: Streaming via SSL after update to v1.40.0

### ISSUE THREAD

**Issue by andyundso** [netdata/netdata#15221]
### Bug description

We run a setup with about twenty children sending data to one parent.

For background: I opened [an issue in the community forum](https://community.netdata.cloud/t/childs-unable-to-connect-to-parent-node/4030) earlier this year when I added two new nodes where the streaming did not work. The update to v1.39.1 helped. Now the two new children work, but every other child fails.

On the children, I see the following:

```
2023-06-20 13:22:54: netdata INFO  : SNDR[child-01 : STREAM child-01: connecting to 'netdata.example.com:19999' (default port: 19999)...
2023-06-20 13:22:54: netdata ERROR : SNDR[child-01 : STREAM child-01 [send to netdata.example.com:19999]: remote netdata does not respond.
```

No message on the parent.

Connection goes over public internet, traffic is encrypted using a self-signed certificate. There is a mix of IPv4 and IPv6 connections, the parent node accepts both but some older nodes still use only IPv4, while all newer ones have both IPv4 and IPv6 enabled, in case this is relevant.

When I deactivate SSL, connection works.

I can confirm the children have a connection to the parent:

```
root@child-01:/var/lib/netdata# telnet -4 netdata.example.com 19999
Trying 1.1.1.1...
Connected to netdata.example.com.
Escape character is '^]'.
^CConnection closed by foreign host.
root@child-01:/var/lib/netdata# telnet -6 netdata.example.com 19999
Trying 2606:4700:4700::1111...
Connected to netdata.example.com.
Escape character is '^]'.
^CConnection closed by foreign host.
```

Connection configuration of the parent:

```
[web]
    ssl key = /etc/netdata/ssl/cert.key
    ssl certificate = /etc/netdata/ssl/cert.crt
    bind to = *:19999=streaming^SSL=force 127.0.0.1:19998=dashboard|badges|management|netdata.conf|registry^SSL=optional
```

Stream configuration of the children:

```
[stream]
    enabled = yes
    destination = netdata.example.com:19999:SSL
    api key = 8D6C6CD7-0B8E-4860-B6E6-78CB27D7A344
    CAfile = /etc/netdata/ssl/cert.pem
    ssl skip certificate verification = yes
    timeout seconds = 60
    send charts matching = *
    buffer size bytes = 1048576
    reconnect delay seconds = 60
    initial clock resync iterations = 60
```

### Expected behavior

The children should be able to connect to the parent using an encrypted connection to send their metrics.

### Steps to reproduce

Described in the bug description.

### Installation method

manual setup of official DEB/RPM packages

### System info

```shell
Parent:

Linux netdata 5.10.0-21-cloud-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux
/etc/cloud-release:ID="genericcloud"
/etc/cloud-release:VERSION="20230124-1270"
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="11"
/etc/os-release:VERSION="11 (bullseye)"
/etc/os-release:VERSION_CODENAME=bullseye
/etc/os-release:ID=debian

Example child:

Linux child-01 4.19.0-18-cloud-amd64 #1 SMP Debian 4.19.208-1 (2021-09-29) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 10 (buster)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="10"
/etc/os-release:VERSION="10 (buster)"
/etc/os-release:VERSION_CODENAME=buster
/etc/os-release:ID=debian
```

### Netdata build info

```shell
Parent:
Version: netdata v1.40.0
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-option-checking' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--disable-dependency-tracking' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -ffile-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -ffile-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:  
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
    HTTPD (h2o):                YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    debugfs:                 YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
Debug/Developer Features:
    Trace Allocations:       NO

Example child:
Version: netdata v1.40.0
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--disable-dependency-tracking' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:  
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         NO
    HTTPD (h2o):                YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    debugfs:                 YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
Debug/Developer Features:
    Trace Allocations:       NO
```

**Comment by ilyam8** [comment]
Hey, @andyundso.

> There is a mix of IPv4 and IPv6 connections, the parent node accepts both 
>
> bind to = [::]:19999=streaming^SSL=force 127.0.0.1:19998=dashboard|badges|management|netdata.conf|registry^SSL=optional

My understanding is that the parent accepts only IPv6 connections (listens on all interfaces), not IPv4 (listens on 127.0.0.1:19998).

**Comment by andyundso** [comment]
@ilyam8 sorry, posted a version of the configuration where I was still experimenting. I updated it in the original issue description. Just double-checked, but same issue when using `*:19999=streaming^SSL=force`.

**Comment by ilyam8** [comment]
I did some tests and it looks like the problem is connection timeout (IP family doesn't matter). I added delay for all outgoing packets to the children instance (`sudo tc qdisc add dev ens18 root netem delay 50ms 10ms`) and get the same problem (remote netdata does not respond) with enabled SSL. No problems without SSL.

**Comment by ilyam8** [comment]
Fixes in #15222. Thanks for reporting, @andyundso!

**Comment by andyundso** [comment]
Hi @ilyam8 

I updated my systems to the latest nightly and can confirm that streaming with SSL works again!

Many thanks for the quick fix.
